src/python/gem5/resources/downloader.py

```diff
@@ -368,3 +380,19 @@ def safe_extract(
 
                 safe_extract(f, unpack_to)
             os.remove(download_dest)
+
+
+def _file_uri_to_path(uri: str) -> Optional[Path]:
+    """
+    Converts a file URI to a local path.
+
+    :param uri: The file URI to convert.
+
+    :returns: The path to the file.
```

Comment:
```suggestion
    If the URI uses the File scheme (e.g, `file://host/path`) then
    a Path object for the local path is returned, otherwise None.
    
    **Note:** Only files from localhost are permitted. An exception
    is thrown otherwise.

    :param uri: The file URI to convert.

    :returns: The path to the file.
```